### DISCUSSION TITLE
Is it possible to disable automatic timestamp operations per create/update call?

### DISCUSSION THREAD

**Discussion by AlexStansfield** [sensedeep/dynamodb-onetable#445] [Q&A] [answered]
I'm migrating all the data from old tables to shiny new OneTable tables and wanted to keep the `createdAt` and `updatedAt` as in the original data.

I read through the docs but didn't see anything, I was wondering if there was a parameter option when making a create/update that would not automatically set the timestamps but instead use whatever was supplied in the properties.

**Comment by mobsense** [comment]
Let me check if adding explicit created / updated timestamps via properties superceds the auto calculated ones.

**Comment by mobsense** [comment, marked as answer]
You can provide params.timestamps == false on your create/update API and it will NOT override the timestamps provided via properties.

If using TypeScript, the Model.d.ts is missing the definition for this. You need to add the following to OneParams:

        timestamps?: boolean

**Reply by AlexStansfield** [reply]
Wow, I've no idea how I missed that. Thank you so much!

**Comment by mobsense** [comment]
Alex, separate issue and urgent.

There is an urgent issue with your streams support. We cannot use the Converter.unmarshall as it created a V2 SDK hard dependency. We need to push out a 2.6.1 that disables this so V3 customers can use 2.6.0. Can you revise your pull requests to not use that dependency?

**Reply by AlexStansfield** [reply]
sure thing, will work on it today.

**Reply by dev-embedthis** [reply]
Thank you!  In 2.6.1 we just added an exception in Table.js and changed jest.config* to ignore streams*.ts.

You will want to revert both these.

**Reply by AlexStansfield** [reply]
I've added a PR - https://github.com/sensedeep/dynamodb-onetable/pull/448

**Reply by AlexStansfield** [reply]
just spotted a linting issue, give me a moment
